ksql-engine/src/test/java/io/confluent/ksql/function/udf/string/SplitKudfTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.confluent.ksql.function.udf.string;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+
+public class SplitKudfTest {
+  private final static SplitKudf splitUdf = new SplitKudf();
+
+  @Test
+  public void shouldReturnNullOnAnyNullParameters() {
+    assertThat(splitUdf.split(null, ""), is(nullValue()));
+    assertThat(splitUdf.split("", null), is(nullValue()));
+    assertThat(splitUdf.split(null, null), is(nullValue()));
+  }
+
+  @Test
+  public void shouldReturnOriginalStringOnNotFoundDelimiter() {
+    assertThat(splitUdf.split("", "."), contains(""));
+    assertThat(splitUdf.split("x-y", "."), contains("x-y"));
+  }
+
+  @Test
+  public void shouldSplitStringByGivenAnEmptyDelimiter() {
+    assertThat(splitUdf.split("", ""), contains(""));
+    assertThat(splitUdf.split("x-y", ""), contains("x", "-", "y"));
+  }
+
+  @Test
+  public void shouldSplitStringByGivenAStringDelimiter() {
+    assertThat(splitUdf.split("x-y", "-"), contains("x", "y"));
+    assertThat(splitUdf.split("x-y", "x"), contains("", "-y"));
+    assertThat(splitUdf.split("x-y", "y"), contains("x-", ""));
+    assertThat(splitUdf.split("a.b.c.d", "."), contains("a", "b", "c", "d"));
+
+    // Add empty spaces if the delimiter is at the beginning, end, or they are contiguous
```

Comment:
nit: maybe put these into a different test so they have a nice name. Makes it easier to just read the tests to see what it does/doesn't do

Reply:
Moved to 
shouldSplitAndAddEmptySpacesIfDelimiterIsFoundAtTheBeginningOrEnd()
and 
shouldSplitAndAddEmptySpacesIfDelimiterIsFoundInContiguousPositions()